### PR TITLE
Add external classification

### PR DESCRIPTION
**Pull request by PawelPodkalicki**

### CODE REVIEW

File: app/Http/Controllers/ClassificationController.php [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+    /** @var ClassifierExternalSignatureVerifier */
+    private $signatureVerifier;
+
+    public function __construct(ClassifierExternalSignatureVerifier $signatureVerifier)
+    {
+        $this->signatureVerifier = $signatureVerifier;
+    }
+
+    public function updateClassification(string $classifier, Request $request): JsonResponse
+    {
```
Comment (m-pilarczyk) [inline app/Http/Controllers/ClassificationController.php:54]: Check the classifier name

File: app/Utilities/ClassifierExternalKeywordsSerializer.php [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        if (self::isAssocAndNotEmpty($array)) {
+            ksort($array);
+        } else {
+            sort($array);
+        }
+    }
+
+    private static function isAssocAndNotEmpty(array $arr): bool
+    {
+        if ([] === $arr) {
```
Comment (m-pilarczyk) [inline app/Utilities/ClassifierExternalKeywordsSerializer.php]: empty

**Review by m-pilarczyk — CHANGES_REQUESTED** [review]

File: app/Utilities/ClassifierExternalKeywordsSerializer.php (hunk 2 above)
Comment (PawelPodkalicki) [reply app/Utilities/ClassifierExternalKeywordsSerializer.php]: Changed to empty.

File: app/Http/Controllers/ClassificationController.php (hunk 1 above)
Comment (PawelPodkalicki) [reply app/Http/Controllers/ClassificationController.php:54]: Added classifier name validation.

**Review by m-pilarczyk — APPROVED** [review]

**Pull request merged**